conans/client/conan_api.py

```diff
@@ -144,11 +144,21 @@ def instance_remote_manager(requester, client_cache, user_io, _client_version,
     @staticmethod
     def factory(interactive=None):
         """Factory"""
-
-        use_color = get_env("CONAN_COLOR_DISPLAY", 1)
-        if use_color and hasattr(sys.stdout, "isatty") and sys.stdout.isatty():
+        # Respect color env setting or check tty if unset
+        color_set = "CONAN_COLOR_DISPLAY" in os.environ
+        if ((color_set and get_env("CONAN_COLOR_DISPLAY", 1))
+                or (not color_set
```

Comment:
I don't fully get this change. The CONAN_COLOR_DISPLAY is obtained with a default of 1 in case it is not defined. It is just a way to disable color, but I am not sure this change is really necessary for this Pycharm issue.

Reply:
So the existing functionality allows you to turn off color, but you only get color if conan determines you are in tty. Because many terminals fail the is tty check but still can have color my changes allow the user to also request color regardless of the tty check.

**Existing behaviour:**  

user request|is tty|not tty
-------------|-------|-------
user requests no color `CONAN_COLOR_DISPLAY=0`|no color|no color
user doesn't specify|color|no color

**New behaviour:**  

user request|is tty|not tty
-------------|-------|-------
user requests no color `CONAN_COLOR_DISPLAY=0`|no color|no color
user requests color `CONAN_COLOR_DISPLAY=1`|color|color
user doesn't specify|color|no color

The above is required to get conan to attempt color when run from pycharms run configurations. The pycharm specific secion:  

```python
if get_env("PYCHARM_HOSTED"):
    convert = False
    strip = False
else:
    convert = None
    strip = None
import colorama
colorama.init(convert=convert, strip=strip)
```

Is required to get the color to actually work as it's not a normal terminal.